R/object_name_linters.R

```diff
@@ -283,7 +284,7 @@ loweralnum <- rex(one_of(lower, digit))
 upperalnum <- rex(one_of(upper, digit))
 
 style_regexes <- list(
-  "symbols"     = rex(start, maybe("."), zero_or_more(none_of(alnum)), end),
+  "symbols"     = rex(start, zero_or_more(none_of(alnum)), end),
```

Comment:
The empty name can occur here because symbols like `<-` are stripped before passing on to the regex.

Reply:
this feels somewhat hacky... but OK for now.

Reply:
The reason is the other styles should allow e.g. `snake_case<-` as a function name

Reply:
As long as applying those rules in the same order always works, we're OK, I'm just not 100% sure of that.